bittensor/axons/__init__.py

```diff
@@ -0,0 +1,98 @@
+# The MIT License (MIT)
+# Copyright © 2021 Yuma Rao
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+# documentation files (the “Software”), to deal in the Software without restriction, including without limitation 
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, 
+# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of 
+# the Software.
+
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION 
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+# DEALINGS IN THE SOFTWARE.
+
+import bittensor
+import argparse
+import copy
+from munch import Munch
+
+from . import _axon
+
+class axon():
+
+    def __new__(
+            cls, 
+            config: Munch = None, 
+            wallet: 'bittensor.wallet.Wallet' = None,
+            local_port: int = None,
+            local_ip: str =  None,
+            max_workers: int = None, 
+            forward_processing_timeout:int = None,
+            backward_processing_timeout:int = None,
+            **kwargs
+        ):
+        # config for the wallet and nucleus sub-objects.
+        if config == None:
+            config = axon.default_config()
+        config = copy.deepcopy(config); bittensor.config.Config.update_with_kwargs(config, kwargs )
+        config.axon.local_port = local_port if local_port != None else config.axon.local_port
+        config.axon.local_ip = local_ip if local_ip != None else config.axon.local_ip
+        config.axon.max_workers = max_workers if max_workers != None else config.axon.max_workers
+        config.axon.forward_processing_timeout = forward_processing_timeout if forward_processing_timeout != None else config.axon.forward_processing_timeout
+        config.axon.backward_processing_timeout = backward_processing_timeout if backward_processing_timeout != None else config.axon.backward_processing_timeout
+        axon.check_config( config )
+
+        # Wallet: Holds you hotkey keypair and coldkey pub, which can be used to sign messages 
+        # and subscribe to the chain.
+        if wallet == None:
+            wallet = bittensor.wallet.Wallet( config = config )
+        wallet = wallet
```

Comment:
This overall looks good to me, just a heads up though, when using __new__ in python and you instantiate the axon, it's going to call __new__ and then call __init__. Just a heads up!

Reply:
Right, but because the __new__ call on the sub Axon is not implemented, this is a no-op right?